packages/block-editor/src/store/private-actions.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * WordPress dependencies
+ */
+import { Platform } from '@wordpress/element';
+
+/**
+ * A list of private/experimental block editor settings that
+ * should not become a part of the WordPress public API.
+ * BlockEditorProvider will remove these settings from the
+ * settings object it receives.
+ *
+ * @see https://github.com/WordPress/gutenberg/pull/46131
+ */
+const privateSettings = [ '__unstableInserterMediaCategories' ];
+
+/**
+ * Action that updates the block editor settings and
+ * conditionally preserves the experimental ones.
+ *
+ * @param {Object}  settings                  Updated settings
+ * @param {boolean} stripExperimentalSettings Whether to strip experimental settings.
+ * @return {Object} Action object
+ */
+export function __experimentalUpdateSettings(
```

Comment:
This one was introduced in #47319 as another locked action.

I've moved it to the `private-actions` file, which is a bit easier to handle compared to specifying every private action/selector in the index file.

Reply:
I used the __experimental prefix to differentiate it from the unprefixed `updateSettings` action. In theory the unlocked store could shadow the stable action with a private one with the same name, but I worry about hard to identify bugs where the private action isn't registered for some reason and you call the public one without realizing and without a clear error message.

Reply:
I agree that we can skip the `__experimentalUpdateSettings` for now and discuss/explore it separately. For newly created actions, selectors, etc.. I think it's good to skip the prefix.